newIDE/app/src/Utils/GDevelopServices/Project.js

```diff
@@ -0,0 +1,271 @@
+// @flow
+import axios from 'axios';
+import {
+  GDevelopProjectApi,
+  GDevelopProjectResourcesStorage,
+} from './ApiConfigs';
+import { type AuthenticatedUser } from '../../Profile/AuthenticatedUserContext';
+
+export const CLOUD_PROJECT_NAME_MAX_LENGTH = 50;
+
+const projectResourcesClient = axios.create({
+  withCredentials: true,
+});
+
+type CloudProject = {|
+  id: string,
+  name: string,
+  createdAt: string,
+  currentVersion?: string,
+  deletedAt?: string,
+|};
+
+export type CloudProjectWithUserAccessInfo = {|
+  ...CloudProject,
+  lastModifiedAt: string,
+|};
+
+const refetchCredentialsForProjectAndRetryIfFailed = async <T>(
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string,
+  apiCall: () => Promise<T>
+): Promise<T> => {
+  try {
+    const response = await apiCall();
+    return response;
+  } catch (error) {
+    if (error.response && error.response.status === 403) {
+      await getCredentialsForCloudProject(authenticatedUser, cloudProjectId);
+      const response = await apiCall();
+      return response;
+    }
+    throw error;
+  }
+};
+
+const getVersionIdFromPath = (path: string): string => {
+  let cleanedPath = path;
+  const searchParamsStartIndex = path.indexOf('?');
+  if (searchParamsStartIndex >= 0) {
+    cleanedPath = path.substring(0, searchParamsStartIndex);
+  }
+  const filenameStartIndex = cleanedPath.lastIndexOf('/') + 1;
+  const filenameEndIndex = cleanedPath.indexOf('.zip');
+  return path.substring(filenameStartIndex, filenameEndIndex);
+};
+
+export const getCredentialsForCloudProject = async (
+  authenticatedUser: AuthenticatedUser,
+  cloudProjectId: string
+): Promise<boolean> => {
+  const { getAuthorizationHeader, firebaseUser } = authenticatedUser;
+  if (!firebaseUser) return false;
+
+  const { uid: userId } = firebaseUser;
+  const authorizationHeader = await getAuthorizationHeader();
+  const response = await axios.get(
+    `${GDevelopProjectApi.baseUrl}/project/${cloudProjectId}/action/authorize`,
+    {
+      headers: { Authorization: authorizationHeader },
+      params: { userId },
+      validateStatus: status => true,
+      withCredentials: true, // Necessary to save cookie returned by the server.
```

Comment:
do you need to specify withCredentials for all those calls? You can't use the `projectResourcesClient` you've created?

Reply:
I cleaned all the uses of axios is the file! Thanks for giving me the motivation to do it 😅

Reply:
🚀 🎸